module-audio/board/rt1051/bellpx/BellPxAudioCodec.cpp

```diff
@@ -124,67 +124,40 @@ namespace audio
 
     void BellPxAudioCodec::InStart()
     {
-        sai_transfer_format_t sai_format;
+        sai_transceiver_t saiConfig;
         auto audioCfg = bsp::audio::AudioConfig::get();
 
-        /* Configure the audio format */
-        sai_format.bitWidth           = saiInFormat.bitWidth;
-        sai_format.channel            = 0U;
-        sai_format.sampleRate_Hz      = saiInFormat.sampleRate_Hz;
-        sai_format.masterClockHz      = audioCfg->mclkSourceClockHz;
-        sai_format.isFrameSyncCompact = false;
-        sai_format.protocol           = audioCfg->config.protocol;
-        sai_format.stereo             = saiInFormat.stereo;
-#if defined(FSL_FEATURE_SAI_FIFO_COUNT) && (FSL_FEATURE_SAI_FIFO_COUNT > 1)
-        sai_format.watermark = FSL_FEATURE_SAI_FIFO_COUNT / 2U;
-#endif
-
         SAI_TransferRxCreateHandleEDMA(BELL_AUDIOCODEC_SAIx,
                                        &rxHandle,
                                        rxAudioCodecCallback,
                                        this,
                                        reinterpret_cast<edma_handle_t *>(audioCfg->rxDMAHandle->GetHandle()));
-
-        SAI_TransferRxSetFormatEDMA(
-            BELL_AUDIOCODEC_SAIx, &rxHandle, &sai_format, audioCfg->mclkSourceClockHz, audioCfg->mclkSourceClockHz);
-
-        DisableIRQ(BELL_AUDIOCODEC_SAIx_RX_IRQ);
-
-        /* Reset SAI Rx internal logic */
-        SAI_RxSoftwareReset(BELL_AUDIOCODEC_SAIx, kSAI_ResetTypeSoftware);
+        /* I2S mode configurations */
+        SAI_GetClassicI2SConfig(&saiConfig, kSAI_WordWidth16bits, kSAI_Stereo, 0U);
+        saiConfig.syncMode    = kSAI_ModeSync;
+        saiConfig.masterSlave = kSAI_Slave;
+        SAI_TransferRxSetConfigEDMA(BELL_AUDIOCODEC_SAIx, &txHandle, &saiConfig);
+        /* set bit clock divider */
+        SAI_RxSetBitClockRate(
+            BELL_AUDIOCODEC_SAIx, audioCfg->mclkSourceClockHz, kSAI_SampleRate44100Hz, kSAI_WordWidth16bits, 2U);
```

Comment:
Shouldn't the sample rate value passed to this function depend on current format's sample rate (`saiInFormat.sampleRate_Hz`)? It seems that previously it was done that way, but maybe now this is applied in different place and I just missed it :wink:

Reply:
Yes, there should be `sampleRate_Hz` from `saiInFormat`, thanks :D

Reply:
I'd also use `saiOutFormat.bitWidth` and `saiOutFormat.stereo` when setting clock rate and getting I2S config - I'm almost 100% sure that our decoder config enforces output in PCM 16-bit stereo format regardless of input file format, but IMO it's better to use these variables and be ready for some potential decoder changes in the future :wink:

Reply:
I'm seeing this now. It was hardcoded because I verified this only with our audio files. I'll rewrite this implementation ;)

Reply:
Done ;)